src/cpp/src/image_generation/models/autoencoder_kl.cpp

```diff
@@ -182,7 +182,16 @@ AutoencoderKL::AutoencoderKL(const std::string& vae_encoder_model,
     compile(device, *extract_adapters_from_properties(properties));
 }
 
-AutoencoderKL::AutoencoderKL(const AutoencoderKL&) = default;
+AutoencoderKL::AutoencoderKL(const AutoencoderKL& rhs) = default;
+
+AutoencoderKL AutoencoderKL::clone() {
+    OPENVINO_ASSERT(!m_decoder_model, "AutoencoderKL must be compiled first. Cannot clone non-compiled model");
```

Comment:
why not? we can clone even when model is not yet compiled?

Reply:
cloning produces new infer request. how can we allow for post-cloning model compilation? there is no ov::CompiledModel to produce infer request from @ilya-lavrenov

Reply:
Extract the CompiledModel from infer request: https://github.com/openvinotoolkit/openvino/blob/994ac20d7fd7fc3ec415dbfa5f0bf6ad886a8162/src/inference/include/openvino/runtime/infer_request.hpp#L328

Reply:
This assertion checks if the model **is compiled**. Only when model is compiled we can get infer request @Wovchena @ilya-lavrenov

Reply:
I see. It's still should be possible to clone by coping ov::Model. But we can postpone it until the time we agree on the rest of the things

Reply:
That's correct

Reply:
Done